### PR TITLE
Bump Python version to 3.10 - 3.12

### PR DESCRIPTION
**Pull request by tanwarsh**
Support for Python version from 3.10 to 3.12.

### CODE REVIEW

File: openfl-tutorials/experimental/workflow/404_Keras_MNIST_with_FedProx.ipynb [hunk 1]
```diff
@@ -51,7 +51,7 @@
    "id": "4dbb89b6",
    "metadata": {},
    "source": [
-    "First we start by installing the necessary dependencies for the workflow interface. Note that this tuorial uses Keras 3, make sure you use python 3.9 or higher."
+    "First we start by installing the necessary dependencies for the workflow interface. Note that this tuorial uses Keras 3, make sure you use python 3.10 or higher."
```
Comment (MasterSkepticista) [inline openfl-tutorials/experimental/workflow/404_Keras_MNIST_with_FedProx.ipynb:54]: OpenFL is assumed to be installed at this point, which would not have happened without python requirements being met. Consider removing mention of python altogether.

**Review by rahulga1 — APPROVED** [review]

File: openfl-tutorials/experimental/workflow/404_Keras_MNIST_with_FedProx.ipynb (hunk 1 above)
Comment (tanwarsh) [reply openfl-tutorials/experimental/workflow/404_Keras_MNIST_with_FedProx.ipynb:54]: The steps mentioned below this comment helps user install Tensorflow, openfl and other required libs. If we are assuming openfl and required lib are installed then we should remove the steps mentioned below as well.

Comment (MasterSkepticista) [reply openfl-tutorials/experimental/workflow/404_Keras_MNIST_with_FedProx.ipynb:54]: `pip` verifies python version requirement for every package that is installed (which is why we mention in `setup.py`). That is an implicit check. If required anyway, it is better to point the user to our official documentation - [Refer to the installation guide for OpenFL](https://openfl.readthedocs.io/en/latest/get_started/installation.html).

My broader point is to increase maintainability of the framework as part of the PR. 
It is one less line to maintain when we bump minimum python to 3.11 next year.

**Review by psfoley — APPROVED** [review]

**Comment by psfoley** [comment]
@tanwarsh The changes look good. Please resolve the merge conflicts. This will need to be brought into `v1.7.x` for consistency with the latest documentation that states support for python 3.10-3.12

**Pull request merged**